user/build.mk

```diff
@@ -71,7 +71,7 @@ endif
 INCLUDE_DIRS += $(MODULE_PATH)/libraries
 
 CFLAGS += -DSPARK_PLATFORM_NET=$(PLATFORM_NET)
-CPPFLAGS += -std=gnu++11
+CPPFLAGS += -std=gnu++14
```

Comment:
C++ 11 to C++ 14.  This seems like too large of a change to be subtly included in this PR.  Is this necessary to switch to now and should we be doing it in it's own PR?

Reply:
This shoudln't cause any issues. C++14 is backwards compatible with C++11 and simply enables additional C++14 features.

Reply:
what is it we need it for? Can we achieve the same in C++11? I can appreciate it's good to upgrade, but I would take the language upgrade at the same time as we upgrade the compiler.

Reply:
`std::make_unique` mainly, as it was forgotten in C++11. FYI we are already enabling C++14 in `services`, `hal` and `system` and gcc 5.3.1 fully supports it (see https://gcc.gnu.org/projects/cxx-status.html#cxx14). There shouldn't be any side-effects or overhead of simply enabling C++14 support in user apps.

I've heard the concerns about this change a couple of times already, but no legitimate reasons against apart from 'what if'.